velox/exec/SortBuffer.cpp

```diff
@@ -285,11 +285,13 @@ void SortBuffer::spillOutput() {
 }
 
 void SortBuffer::prepareOutput(uint32_t maxOutputRows) {
-  VELOX_CHECK_GT(maxOutputRows, 0);
+  VELOX_CHECK_GT(static_cast<vector_size_t>(maxOutputRows), 0);
   VELOX_CHECK_GT(numInputRows_, numOutputRows_);
-
+  const uint64_t batchSizeMaybe = numInputRows_ - numOutputRows_;
   const vector_size_t batchSize =
-      std::min<vector_size_t>(numInputRows_ - numOutputRows_, maxOutputRows);
+      batchSizeMaybe > std::numeric_limits<vector_size_t>::max()
```

Comment:
How about?
```
vector_size_t batchSize = maxOutputRows;
if (numOutputRows_ + batchSize > numInputRows_) {
   batchSize = numInputRows_ - numOutputRows_;
}
...
```

Reply:
numOutputRows_ + batchSize may overflow when numOutputRows_  is UINT64_MAX and numInputRows_ is UINT64_MAX - 1